core/src/bson_type_info.rs

```diff
@@ -92,29 +113,30 @@ pub struct BsonTypeInfo {
 pub struct SimpleTypeInfo {
     pub sql_type: SqlDataType,
     pub non_concise_type: SqlDataType,
-    pub length: Option<u16>,
+    pub length: fn(Option<u16>) -> Option<u16>,
     pub transfer_octet_length: Option<u16>,
-    pub display_size: Option<u16>,
+    pub display_size: fn(Option<u16>) -> Option<u16>,
 }
 
-impl SimpleTypeInfo {
-    const fn new(length: u16, transfer_octet_length: u16, display_size: u16) -> Option<Self> {
-        Some(Self {
+macro_rules! new_simple_type_info {
+    ($length:expr, $transfer_octet_length:expr, $display_size:expr) => {
+        Some(SimpleTypeInfo {
             sql_type: SqlDataType::SQL_WVARCHAR,
             non_concise_type: SqlDataType::SQL_WVARCHAR,
-            length: Some(length),
-            transfer_octet_length: Some(transfer_octet_length),
-            display_size: Some(display_size),
+            length: make_default_attr_func!(Some($length)),
+            transfer_octet_length: Some($transfer_octet_length),
+            display_size: make_default_attr_func!(Some($display_size)),
         })
-    }
-
+    };
+}
+impl SimpleTypeInfo {
     const fn default() -> Option<Self> {
         Some(Self {
             sql_type: SqlDataType::SQL_WVARCHAR,
             non_concise_type: SqlDataType::SQL_WVARCHAR,
-            length: None,
+            length: |max_string_length| max_string_length,
             transfer_octet_length: None,
-            display_size: None,
+            display_size: |max_string_length| max_string_length,
```

Comment:
Talked to Jonathan about this. We figure it is relevant to impose this limit on the stringified types in simple type mode, since those are returned as strings.